lib/DatePickerItem.tsx

```diff
@@ -34,9 +34,10 @@ const DatePickerItem: FC<DatePickerItemProps> = ({
   const currentIndex = useRef(MIDDLE_INDEX);
   const moveDateCount = useRef(0);
   const [mouseDown, setMouseDown] = useState(false);
+  const mouseMoved = useRef(false);
   const moveToTimer = useRef<ReturnType<typeof setTimeout> | void>();
   const [stateTranslateY, setStateTranslateY] = useState(MIDDLE_Y);
-  const [dates, setDates] = useState(iniDates({ step, type, value }));
+  const dates = useRef(iniDates({ step, type, value }));
```

Comment:
Treat `dates` as ref not state. `stateTranslateY` is enough to update UI.
Because, `dates` is changed by side effect of updating  `stateTranslateY`.
(= User Interaction changes `stateTranslateY` first. not `dates`)

Reply:
This might lead to issues with re-rendering when the value prop changes.
If current `stateTranslateY` and `marginTop` are at their initial values, calling `setStateTranslateY` in useEffect won't trigger a re-render.
To make sure the component re-render correctly when `props.value` changes, we should keep using state instead of refs.

Reply:
> This might lead to issues with re-rendering when the value prop changes.
If current stateTranslateY and marginTop are at their initial values, calling setStateTranslateY in useEffect won't trigger a re-render.

That makes sense.
So, I changed to treat `dates` as ref and state both.
(Check the `useStateRef` hook)

Reply:
[0acb474](https://github.com/jin60641/react-mobile-datepicker/pull/3/commits/0acb4749c8908742fb89cf12295a4eff9563a2ad)